### PR TITLE
Translatability

### PR DESCRIPTION
**Pull request by Z3rca**
Allows to translate the mod up to 95-99% into different languages.

**Usage**:
Anyone can create himself based on a Translation Template we provide, a new module, and can push it to nexus , mod db, or steam.

The translation module needs to be loaded after tor has been loaded.

The translation module only contains the Language folder.

Ingame, just switch the language to any other language. Done. 

The template will contain all necessary IDs as an example, so people do not have to look long for IDs only findable in code. 

### Implementation: 

Looks like a massive change but it's probably not a big deal.

The biggest changes: 

**Inky:**
- on every inky story text string are now a "#STR_..." seen which is the only way to reference the according text passage.
- choices can be referenced within context(Knot-name) + "_c0-c..." in the name and only need to be mentioned in the translation 
- minor changes on the logic for the inky stories, so it allows be easier translated. -> Please only have a text + single multiple-choice decision per knot. I could not imagine any case, where it would be relevant to have multiple decision blocks in one knot. 
-in text variables "Its a (cold, hot) day" are translated via new variables introduced, that will be applied to the Textmanager. 

**Career perks**
- Career texts use the constructed string of the tor_strings. 
- In the Tor Strings we also used basic perk texts (adds health points, physical damage, etc.) which do not require effort. 
- Also the referenced strings use Variables instead of hardcoded numbers, for easier on-demand balance changes without affecting translations.
-  The description texts inside the code are only utilized if the perk couldn't be found in the tor_strings XML.

**Other changes:**
- Character creation has now some Utility-method-checks instead of resulting String checks
- dialog lines contain now their translation id, as well as a bazillon of other small methods with text content
-  blank strings have been capsuled into Text Objects to trigger the translation

**Note**
The Career perks still need to be translated and provided with according ids. Piero is currently working on it. It has however nothing to do with the translatability of the mod so I think its ok when its for now left out.

### CODE REVIEW

File: CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs [hunk 1]
```diff
@@ -297,10 +298,10 @@ private void determinetestoutcome()
 
         private bool testresultcondition()
         {
-            var result = "Error.";
-            if (_testResult == "success")
+            var result = new TextObject("Error.");
+            if (_testResult == "success")       //why is this not a simple boolean?
```
Comment (jzz111) [inline CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:302]: probably because it's a text string... "success" is not a boolean

Comment (Z3rca) [reply CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:302]: right above it, there is an assignment of the variable _testResult. the test result is stored as string then we use a string comparision with the same values. A simple boolean might be completely sufficient. If there is no need for it I would just store it as a boolean

Comment (Z3rca) [reply CSharpSourceCode/CampaignMechanics/SpellTrainers/SpellTrainerInTownBehavior.cs:302]: nvm we need the result for the magic test condition, since booleans cant be null . Here it could be that you pre test or post test.  Apart this is all ignoring that we don't use this anyway. I note this down for another clean up round

**Review by hunharibo — APPROVED** [review]
